src/api/commands/escrow/EscrowRefundCommand.ts

```diff
@@ -91,14 +59,83 @@ export class EscrowRefundCommand extends BaseCommand implements RpcCommandInterf
         } as EscrowRequest);
     }
 
+    public async validate(data: RpcRequest): Promise<RpcRequest> {
+        let orderItem;
+        if (data.params.length >= 1) {
+            const orderItemId = data.params[0];
+            if (typeof orderItemId !== 'number' || orderItemId < 0) {
+                throw new InvalidParamException('orderItemId', 'number');
+            }
+
+            let orderItemModel;
+            try {
+                orderItemModel = await this.orderItemService.findOne(orderItemId);
+            } catch (ex) {
+                this.log.error('Error: ' + ex);
+                throw new NotFoundException(orderItemId);
+            }
+            orderItem = orderItemModel.toJSON();
+
+            if (orderItem.status !== OrderStatus.AWAITING_ESCROW) {
+                this.log.error('Order is in invalid state');
+                throw new MessageException('Order is in invalid state');
+            }
+
+            const bid = orderItem.Bid;
+            if (!bid || bid.action !== BidMessageType.MPA_ACCEPT) {
+                this.log.error('No valid information to finalize escrow');
+                throw new MessageException('No valid information to finalize escrow');
+            }
+
+            const listingItem = orderItem.Bid.ListingItem;
+            if (_.isEmpty(listingItem)) {
+                this.log.error('ListingItem not found!');
+                throw new ModelNotFoundException('listingItem');
+            }
+
+            const paymentInformation = orderItem.Bid.ListingItem.PaymentInformation;
+            if (_.isEmpty(paymentInformation)) {
+                this.log.error('PaymentInformation not found!');
+                throw new ModelNotFoundException('paymentInformation');
+            }
+
+            const escrow = orderItem.Bid.ListingItem.PaymentInformation.Escrow;
+            if (_.isEmpty(escrow)) {
+                this.log.error('Escrow not found!');
+                throw new ModelNotFoundException('escrow');
+            }
+
+            const escrowRatio = orderItem.Bid.ListingItem.PaymentInformation.Escrow.Ratio;
+            if (_.isEmpty(escrowRatio)) {
+                this.log.error('EscrowRatio not found!');
+                throw new ModelNotFoundException('escrowRatio');
+            }
+        }
```

Comment:
missing else, should throw missingparamexception

Reply:
also, there's no need for the if else, data.params[0] should always exist and all the checks inside the if should always be run, the if just adds complexity which just caused a bug.